x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,214 @@
+	var icaCallbackData []byte;
+
+	// Send the transaction through SubmitTx to kick off ICA commands -- no ICA callback method name, or callback args needed
+	_, err := k.SubmitTxs(ctx, 
+		hostZone.ConnectionId, 
+		msgs, 
+		types.ICAAccountType_COMMUNITY_POOL_DEPOSIT, 
+		icaTimeoutTimestamp, 
+		icaCallbackId, 
+		icaCallbackData)
+	if err != nil {
+		return errorsmod.Wrapf(types.ErrICATxFailed, "Failed to SubmitTxs, Messages: %v, err: %s", msgs, err.Error())
+	}
+	k.Logger(ctx).Info(fmt.Sprintf("[TransferCommunityPoolTokens] Successfully sent ICA command to kick off remote ibc transfer from community pool deposit ICA"))
+
+	return nil
+}
+
+
+// Transfers all tokens in the Stride-side holding address over to the communityPoolReturnAddress ICA
+func (k Keeper) ReturnAllCommunityPoolTokens(ctx sdk.Context, hostZone types.HostZone) error {
+	// Use bankKeeper to see all coin types and amounts currently in the stride-side holding module address
+	req := banktypes.NewQueryAllBalancesRequest(sdk.AccAddress(hostZone.CommunityPoolHoldingAddress), nil)
+	resp, err := k.bankKeeper.AllBalances(ctx, req)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Couldn't look up balances in holding address")
+	}
+
+	var transferErr error;
+	for _, foundCoin := range resp.Balances {
+		if transferErr = k.TransferCoinToReturn(ctx, hostZone, foundCoin); transferErr != nil {
+			k.Logger(ctx).Error(errorsmod.Wrapf(transferErr, "error in token transfer %+v", foundCoin).Error())
+			continue
+		}
+	}
+
+	return transferErr // will be nil if no errors in loop
+}
+
+
+func (k Keeper) TransferCoinToReturn(ctx sdk.Context, hostZone types.HostZone, coin sdk.Coin) error {
+	memo := ""
+	ibcTransferTimeoutNanos := k.GetParam(ctx, types.KeyIBCTransferTimeoutNanos)
+	timeoutTimestamp := uint64(ctx.BlockTime().UnixNano()) + ibcTransferTimeoutNanos
+
+	// build and send an IBC message for each coin to transfer all back to the hostZone
+	msg := ibctypes.NewMsgTransfer(
+		ibctypes.PortID,
+		hostZone.TransferChannelId,
+		coin,
+		hostZone.CommunityPoolHoldingAddress, // from Stride address, unique to each community pool / hostzone
+		hostZone.CommunityPoolReturnIcaAddress, // to ICA controlled address on foreign hub
+		clienttypes.Height{},
+		timeoutTimestamp,
+		memo,
+	)
+
+	msgTransferResponse, err := k.RecordsKeeper.TransferKeeper.Transfer(sdk.WrapSDKContext(ctx), msg)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Error submitting ibc transfer for %+v", coin)
+	} else {
+		//return errorsmod.Wrapf(errors.New(""), "denom is %s", k.GetDenomOnHostZone(coin.Denom, hostZone))
+		result := fmt.Sprintf("Successfully submitted ibctransfer for %+v with response %+v", 
+				coin, msgTransferResponse)
+		k.Logger(ctx).Info(result)
+
+		// If there was no error in sending the transfer msg, store what the transferred denom will be in callback with amount
+		callbackArgs := types.CommunityPoolReturnTransferCallback{
+			HostZoneId: hostZone.ChainId,
+			DenomStride: coin.Denom,
```

Comment:
DenomStride is only used for logging, so I think you can remove it from the callback data

Reply:
Yep, and in the 3 epoch plan, we don't need to store this at all since the 3rd epoch triggers an ICQ on known denoms instead of needing to remember the amount and denom for the fundcommunitypool call